src/esp32/esp32_wifi.c

```diff
@@ -327,6 +327,31 @@ bool mgos_wifi_dev_sta_setup(const struct mgos_config_wifi_sta *cfg) {
     tcpip_adapter_dhcpc_start(TCPIP_ADAPTER_IF_STA);
   }
 
+  LOG(LL_INFO, ("Setting STA protocol: %s", cfg->protocol));
+  if (strcmp(cfg->protocol, "B") == 0) {
+    r = esp_wifi_set_protocol(WIFI_IF_STA, WIFI_PROTOCOL_11B);
+  } 
+  else if (strcmp(cfg->protocol, "BG") == 0) {
+    r = esp_wifi_set_protocol(WIFI_IF_STA, WIFI_PROTOCOL_11B| WIFI_PROTOCOL_11G);
+  }
+  else if (strcmp(cfg->protocol, "BGN") == 0) {
+    r = esp_wifi_set_protocol(WIFI_IF_STA, WIFI_PROTOCOL_11B| WIFI_PROTOCOL_11G|WIFI_PROTOCOL_11N);
+  }
+  else if (strcmp(cfg->protocol, "BGNLR") == 0) {
+    r = esp_wifi_set_protocol(WIFI_IF_STA, WIFI_PROTOCOL_11B| WIFI_PROTOCOL_11G|WIFI_PROTOCOL_11N|WIFI_PROTOCOL_LR);
+  }
+  else if (strcmp(cfg->protocol, "LR") == 0) {
+    r = esp_wifi_set_protocol(WIFI_IF_STA, WIFI_PROTOCOL_LR);
+  }
+  else {
+    r = esp_wifi_set_protocol(WIFI_IF_STA, WIFI_PROTOCOL_11B| WIFI_PROTOCOL_11G|WIFI_PROTOCOL_11N);
```

Comment:
it sounds like you can use strchr() to build a bitmask.
```
uin32_t protocol = 0;
if (strchr(cfg->protocol, "B) != NULL) protocol |= WIFI_PROTOCOL_11B;
...
r = esp_wifi_set_protocol(WIFI_IF_STA, protocol);
```

Reply:
bitmask created, thanks for the tip!